controllers/tlspolicy_status_updater.go

```diff
@@ -0,0 +1,240 @@
+			{Kind: &machinery.GatewayGroupKind},
+			{Kind: &kuadrantv1alpha1.TLSPolicyGroupKind, EventType: ptr.To(controller.CreateEvent)},
+			{Kind: &kuadrantv1alpha1.TLSPolicyGroupKind, EventType: ptr.To(controller.UpdateEvent)},
+			{Kind: &CertManagerCertificateKind},
+			{Kind: &CertManagerIssuerKind},
+			{Kind: &CertManagerClusterIssuerKind},
+		},
+		ReconcileFunc: t.UpdateStatus,
+	}
+}
+
+func (t *TLSPolicyStatusUpdaterReconciler) UpdateStatus(ctx context.Context, _ []controller.ResourceEvent, topology *machinery.Topology, _ error, s *sync.Map) error {
+	logger := controller.LoggerFromContext(ctx).WithName("TLSPolicyStatusUpdaterReconciler").WithName("Reconcile")
+
+	policies := lo.FilterMap(topology.Policies().Items(), func(item machinery.Policy, index int) (*kuadrantv1alpha1.TLSPolicy, bool) {
+		p, ok := item.(*kuadrantv1alpha1.TLSPolicy)
+		return p, ok
+	})
+
+	store, ok := s.Load(TLSPolicyAcceptedKey)
+	if !ok {
+		logger.Error(errors.New("TLSPolicyAcceptedKey not found, skipping update of tls policy statuses"), "sync map error")
+		return nil
+	}
+
+	isPolicyValidErrorMap := store.(map[string]error)
+
+	for _, policy := range policies {
+		if policy.DeletionTimestamp != nil {
+			logger.V(1).Info("tls policy is marked for deletion, skipping", "name", policy.Name, "namespace", policy.Namespace)
+			continue
+		}
+
+		newStatus := &kuadrantv1alpha1.TLSPolicyStatus{
+			// Copy initial conditions. Otherwise, status will always be updated
+			Conditions:         slices.Clone(policy.Status.Conditions),
+			ObservedGeneration: policy.Status.ObservedGeneration,
+		}
+
+		err := isPolicyValidErrorMap[policy.GetLocator()]
+		meta.SetStatusCondition(&newStatus.Conditions, *kuadrant.AcceptedCondition(policy, err))
+
+		// Do not set enforced condition if Accepted condition is false
+		if meta.IsStatusConditionFalse(newStatus.Conditions, string(gatewayapiv1alpha2.PolicyReasonAccepted)) {
+			meta.RemoveStatusCondition(&newStatus.Conditions, string(kuadrant.PolicyConditionEnforced))
+		} else {
+			enforcedCond := t.enforcedCondition(ctx, policy, topology)
+			meta.SetStatusCondition(&newStatus.Conditions, *enforcedCond)
+		}
+
+		// Nothing to do
+		equalStatus := equality.Semantic.DeepEqual(newStatus, policy.Status)
+		if equalStatus && policy.Generation == policy.Status.ObservedGeneration {
+			logger.V(1).Info("policy status unchanged, skipping update")
+			continue
+		}
+		newStatus.ObservedGeneration = policy.Generation
+		policy.Status = *newStatus
+
+		resource := t.Client.Resource(kuadrantv1alpha1.TLSPoliciesResource).Namespace(policy.GetNamespace())
+		un, err := controller.Destruct(policy)
+		if err != nil {
+			logger.Error(err, "unable to destruct policy")
+			continue
+		}
+
+		_, err = resource.UpdateStatus(ctx, un, metav1.UpdateOptions{})
+		if err != nil {
+			logger.Error(err, "unable to update status for TLSPolicy", "name", policy.GetUID(), "namespace", policy.GetNamespace(), "uid", policy.GetUID())
```

Comment:
The log message at line 67 uses the policy name instead of the object's UID, which I personally think it's easier for debugging. Any particular reason not to do the same here?

Reply:
Nope, I think it's meant to be the name. I'll update 👍